### PR TITLE
MXRT600: Add I2C support

### PR DESCRIPTION
**Pull request by mmahadevan108**
Enable I2C access to FXOS7000 sensor on the mxrt685_evk board. Tested with the sample application samples/sensor/fxos8700, below is the test run log:
 
*** Booting Zephyr OS build zephyr-v2.3.0-1374-geca2b0936c09  ***
AX=  0.076614 AY= -0.181960 AZ=  9.854534 MX= -0.339000 MY= -0.875000 MZ= -0.934000 T= 24.000000
AX= -0.153229 AY= -0.143653 AZ=  9.509768 MX= -0.159000 MY= -0.762000 MZ= -0.940000 T= 23.040000
AX= -0.038308 AY= -0.153229 AZ=  9.528922 MX= -0.153000 MY= -0.738000 MZ= -0.954000 T= 23.040000
AX= -0.028731 AY= -0.143653 AZ=  9.538499 MX= -0.154000 MY= -0.753000 MZ= -0.932000 T= 24.000000
AX= -0.047885 AY= -0.153229 AZ=  9.528922 MX= -0.144000 MY= -0.759000 MZ= -0.949000 T= 24.000000
AX= -0.047885 AY= -0.162806 AZ=  9.538499 MX= -0.148000 MY= -0.754000 MZ= -0.955000 T= 23.040000
AX= -0.028731 AY= -0.162806 AZ=  9.567229 MX= -0.146000 MY= -0.760000 MZ= -0.942000 T= 23.040000
AX= -0.019154 AY= -0.172383 AZ=  9.528922 MX= -0.144000 MY= -0.767000 MZ= -0.948000 T= 23.040000
AX= -0.038308 AY= -0.134076 AZ=  9.557653 MX= -0.145000 MY= -0.743000 MZ= -0.952000 T= 24.000000
AX= -0.047885 AY= -0.153229 AZ=  9.528922 MX= -0.150000 MY= -0.749000 MZ= -0.975000 T= 24.000000
AX= -0.047885 AY= -0.172383 AZ=  9.548076 MX= -0.166000 MY= -0.754000 MZ= -0.971000 T= 24.000000
AX=  0.000000 AY= -0.172383 AZ=  9.519345 MX= -0.156000 MY= -0.763000 MZ= -0.939000 T= 23.040000
AX= -0.047885 AY= -0.181960 AZ=  9.519345 MX= -0.156000 MY= -0.764000 MZ= -0.957000 T= 24.000000
AX= -0.028731 AY= -0.172383 AZ=  9.528922 MX= -0.152000 MY= -0.754000 MZ= -0.975000 T= 23.040000
AX= -0.067038 AY= -0.153229 AZ=  9.557653 MX= -0.154000 MY= -0.761000 MZ= -0.950000 T= 24.000000
AX= -0.019154 AY= -0.143653 AZ=  9.509768 MX= -0.163000 MY= -0.761000 MZ= -0.965000 T= 24.000000
AX= -0.047885 AY= -0.134076 AZ=  9.538499 MX= -0.148000 MY= -0.761000 MZ= -0.967000 T= 24.000000
AX= -0.038308 AY= -0.143653 AZ=  9.528922 MX= -0.153000 MY= -0.749000 MZ= -0.960000 T= 24.000000
AX= -0.038308 AY= -0.153229 AZ=  9.519345 MX= -0.145000 MY= -0.746000 MZ= -0.953000 T= 23.040000
AX= -0.019154 AY= -0.134076 AZ=  9.519345 MX= -0.166000 MY= -0.749000 MZ= -0.938000 T= 24.000000

### CODE REVIEW

**Comment by mmahadevan108** [comment]
cc @MaureenHelm @dleach02 @agansari

**Review by agansari — CHANGES_REQUESTED** [review]
> Enable I2C access to FXOS7000 sensor on the mxrt685_evk board. Tested with the sample application samples/sensor/fxos8700, below is the test run log:

Please add the board to whitelist in: _samples/sensor/fxos8700/sample.yaml_

File: boards/arm/mimxrt685_evk/Kconfig.defconfig [hunk 1]
```diff
@@ -28,4 +28,8 @@ config GPIO_MCUX_LPC_PORT1
 
 endif # GPIO_MCUX_LPC
 
+config FXOS8700_DRDY_INT1
+	default y
+	depends on FXOS8700
```
[the remark this answers is not in the document]
Comment (MaureenHelm) [reply boards/arm/mimxrt685_evk/Kconfig.defconfig:33]: Probably a copy/paste from another board that does this incorrectly too

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Looks good except for the bits caught by @agansari

File: boards/arm/mimxrt685_evk/Kconfig.defconfig (hunk 1 above)
[the remark this answers is not in the document]
Comment (mmahadevan108) [reply boards/arm/mimxrt685_evk/Kconfig.defconfig]: Thank you, I have changed this.

**Comment by mmahadevan108** [comment]
> > Enable I2C access to FXOS7000 sensor on the mxrt685_evk board. Tested with the sample application samples/sensor/fxos8700, below is the test run log:
> 
> Please add the board to whitelist in: _samples/sensor/fxos8700/sample.yaml_

I have updated the whitelist and README files.

**Review by agansari — APPROVED** [review]

**Comment by mmahadevan108** [comment]
I realized that the same I2C connected to FXOS7000 sensor is also connected to arduino I2C pins. I have updated this PR to include 
"arduino_i2c: &flexcomm2"  instead of just "&flexcomm2"

**Review by MaureenHelm — APPROVED** [review]
> I realized that the same I2C connected to FXOS7000 sensor is also connected to arduino I2C pins. I have updated this PR to include
"arduino_i2c: &flexcomm2" instead of just "&flexcomm2"

Nice catch, thank you

File: boards/arm/mimxrt685_evk/Kconfig.defconfig (hunk 1 above)
[the remark this answers is not in the document]
Comment (MaureenHelm) [reply boards/arm/mimxrt685_evk/Kconfig.defconfig]: Fixed the other boards in #27294. Good catch @agansari

**Pull request merged**